### ISSUE TITLE
py 3.6 on windows (only) test_get_instance_* (5 tests) failing

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandi-cli#193]
while building conda package:
https://github.com/conda-forge/dandi-feedstock/pull/5

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=196889&view=logs&j=171a126d-c574-5c8c-1269-ff3b989e923d&t=1183ba29-a0b5-5324-8463-2a49ace9e213&l=863

e.g.
```
___________________________ test_get_instance_dandi ___________________________

    @responses.activate
    def test_get_instance_dandi():
        responses.add(
            responses.GET,
            "https://dandiarchive.org/server-info",
            json={
                "version": "1.0.0",
                "cli-minimal-version": "0.5.0",
                "cli-bad-versions": [],
                "services": {
                    "girder": {"url": "https://girder.dandi"},
                    "webui": {"url": "https://gui.dandi"},
                    "api": {"url": "https://publish.dandi/api"},
                    "jupyterhub": {"url": "https://hub.dandi"},
                },
            },
        )
>       assert get_instance("dandi") == dandi_instance(
            girder="https://girder.dandi",
            gui="https://gui.dandi",
            redirector="https://dandiarchive.org",
            api="https://publish.dandi/api",
        )
E       AssertionError: assert dandi_instanc...hive.org/api') == dandi_instanc...sh.dandi/api')
E         At index 0 diff: 'https://girder.dandiarchive.org' != 'https://girder.dandi'
E         Full diff:
E         - dandi_instance(girder='https://girder.dandi', gui='https://gui.dandi', redirector='https://dandiarchive.org', api='https://publish.dandi/api')
E         + dandi_instance(girder='https://girder.dandiarchive.org', gui='https://gui.dandiarchive.org', redirector='https://dandiarchive.org', api='https://publish.dandiarchive.org/api')
E         ?                                            +++++++++++                         +++++++++++                                                                    +++++++++...
E         
E         ...Full output truncated (1 line hidden), use '-vv' to show

..\_test_env\lib\site-packages\dandi\tests\test_utils.py:170: AssertionError
------------------------------ Captured log call ------------------------------
WARNING  dandi:utils.py:565 Request to https://dandiarchive.org failed ('HTTPHeaderDict' object has no attribute 'get_all')
WARNING  dandi:utils.py:567 Using hard-coded URLs
____________________________ test_get_instance_url ____________________________
...
```

any quick idea @jwodder ?

**Comment by jwodder** [comment]
@yarikoptic This is apparently due to a mix of incompatible versions of `responses`, `requests`, and `urllib3`: https://github.com/getsentry/responses/issues/292

**Comment by yarikoptic** [comment]
well, in https://github.com/getsentry/responses/issues/292#issuecomment-670760655 "fix for me was to force upgrade urllib3 to 1.25.10", which is what we had according to "setup" step: https://dev.azure.com/conda-forge/84710dde-1620-425b-80d0-4cf5baca359d/_apis/build/builds/196889/logs/36 and requests also recent 2.24.0  ... but then in build step it is indeed older versions are installed (apparently `setup` is not really the `build` environment :-( bleh).  I will try to demand `requests >= 2.22.0`

**Comment by yarikoptic** [comment]
and that seemed to help! merging the feedstock recipe now. Thank you @jwodder for digging that issue out!
